docs/products/kafka/concepts/upgrade-procedure.rst

```diff
@@ -0,0 +1,253 @@
+Apache Kafka® upgrade procedure
+===============================
+
+One of the benefits of using a managed service like Aiven for Apache Kafka® is the automated upgrade procedure. 
+
+The upgrade procedure is executed during:
+    
+* maintenance updates
+* plan changes
+* cloud region migrations
+* manual node replacements performed by an Aiven operator
+    
+All the above operations involve creating new broker nodes to replace existing ones.
+
+Upgrade procedure steps
+---------------------------
+
+The following set of steps are executed during an upgrade procedure, the example shows a possible scenario for a 3-node Apache Kafka service.
+
+.. mermaid::
```

Comment:
Is there a possibility for dark mode in mermaid? It's not awfully bad in dark mode, but visibly off

Reply:
As of now not